ClinGen/clincoded
# Connects to #1077 kl schema case control

This PR contains changes in schema, test data and handling for case control, including:
 1. Created schema files `caseCohort.json`, `controlCohort.json` and `caseControl.json` in dir `/src/clincoded/schemas/`, and generated related test data files `caseCohort.json`, `controlCohort.json` and `caseControl.json` in dir `/src/clincoded/tests/data/inserts`.
 2. Added data property `caseControlStudies` in schema file `/src/clincoded/schemas/annotation.json` and edited related test data in file `/src/clincoded/tests/data/inserts/annotation.json`.
 3. Added new classes `CaseCohort`, `ControlCohort` and `CaseControl` and edited classes `Annotation` and ` Gdm` in file `/src/clincoded/types/__init__.py` to handle linked and reversely linked objects so that 3 new objects are embedded in GDM.
 4. Added items "caseCohort", "controlCohort" and "caseControl" in file `src/clincoded/loadxl.py` to handle data loading.
 5. Generated new dump files.

**Test**
* Object caseCohort
  - Go to [http://localhost:6543/casecohort/](http://localhost:6543/casecohort/) and expect to see
![screen shot 2016-10-18 at 1 51 49 pm](https://cloud.githubusercontent.com/assets/9206696/19495886/0e6a3072-953a-11e6-9606-b2cadf5ea011.png)
  - Click "Case Cohort 001" and expect to see case cohort data.

* Object controlCohort
  - Go to [http://localhost:6543/controlcohort/](http://localhost:6543/controlcohort/) and expect to see
![screen shot 2016-10-18 at 1 56 42 pm](https://cloud.githubusercontent.com/assets/9206696/19496067/bab6ba76-953a-11e6-8e11-a2aca8ba8e13.png)
  - Click "Control Cohort 01" and expect to see control cohort data.

* Object caseControl
  - Go to [http://localhost:6543/casecontrol/](http://localhost:6543/casecontrol/) and expect to see
![screen shot 2016-10-18 at 1 58 03 pm](https://cloud.githubusercontent.com/assets/9206696/19496117/f84be17c-953a-11e6-8097-63cec34c96dc.png)
  - Click the uuid and expect to see case control data.

* Embedded in GDM
  - Go to [http://localhost:6543/gdm/9ffd7c1e-16d9-11e5-b283-60f81dc5b05a/](http://localhost:6543/gdm/9ffd7c1e-16d9-11e5-b283-60f81dc5b05a/) and expect to see `caseCohort` and `controlCohort` embedded in `caseControlStudies` which is embedded in `annotations` in the GDM. 

**END of Test**
